File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <set>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    set<int> s;
    
    for(int i=0; i<nums.size(); i++) {
        s.insert(nums[i]);
    }
    int get = nums.size()/2;
    
    // if(s.size()>get) {
    //     return get;
    // } else {
    //     return s.size();
    // }
    return min(s.size(), nums.size()/2);
    
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        // 3 3 
        // 2 3
        // 3개
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        
        for(int i : nums) {
            map.put(i, map.getOrDefault(i,0)+1);
        }
        
        if(map.size()>= n/2) {
            return n/2;
        } else {
            return map.size();
        }
        
    }
}